core/schema/host.go

```diff
@@ -413,17 +414,29 @@ func loadDirectoryGitIgnorePatterns(ctx context.Context, parentPath string, host
 				{Name: "noGitAutoIgnore", Value: dagql.Boolean(true)},
 			},
 		},
+		dagql.Selector{
+			Field: "__gitIgnoreFor",
+		},
 	)
 	if err != nil {
 		return nil, fmt.Errorf("failed to load directory ignore patterns: %w", err)
 	}
 
-	gitIgnorePatterns, err := core.LoadGitIgnoreInDirectory(ctx, directory)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load .gitignore in context directory %q: %w", parentPath, err)
+	for i := 1; i <= result.Len(); i++ {
+		entry, err := result.Nth(i)
```

Comment:
`dagql.ResultArray` is just an array type, so you can do `for _, entry := range result`

Reply:
Applied in [7ff0141](https://github.com/dagger/dagger/pull/10853/commits/7ff0141df57caa39e12fadd9c25b19c5062431bb)